File: Python/1023. 形成字符串的最短路径.py
# If you need to import additional packages or classes, please import here.
def is_substr(sub, s):
    m = len(sub)
    n = len(s)
    i, j = 0, 0

    while i < m and j < n:
        if(sub[i] == s[j]):
            i += 1
            j += 1
        else:
            j += 1
    return i  == m
            
def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    source = input()
    target = input()
        
    res = 0
    i, j = 0, 1
    m = len(target)
    while i < m :
        if is_substr(target[i:j], source):
            if j < m:
                j += 1
            else:
                res += 1
                break
        else:
            if j-i==1:
                res = -1
                break
            else:
                res += 1
                i = j - 1
    print(res)
if __name__ == "__main__":
    func()

File: C++/1023. 形成字符串的最短路径.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
bool isSubstr(string sub, string s) {
    int m = sub.size();
    int n = s.size();
    int i = 0;
    int j = 0;
    
    while(i < m && j < n) {
        if(sub[i] == s[j]) {
            i++;
        }
        j++;
    }
    return i == m;
}

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    string source, target;
    cin>>source;
    cin>>target;
    int i = 0;
    int j = 1;
    int m = target.size();
    int res = 0;
    while(i < m) {
        if(isSubstr(target.substr(i, j - i), source)) {
            if(j < m) {
                j++;
            } else {
                res++;
                break;
            }
        } else {
            if(j - i == 1) {
                res = -1;
                break;
            } else {
                i = j - 1;
                res++;
            }
        }
    }
    cout<<res<<endl;
    
    return 0;
}
